server/src/com/thoughtworks/go/server/websocket/ConsoleLogSender.java

```diff
@@ -92,6 +92,11 @@ public void process(final SocketEndpoint webSocket, JobIdentifier jobIdentifier,
             // loop before we've seen the last content update
             if (isRunningBuild) sendLogs(webSocket, streamer, jobIdentifier);
 
+            //send the remaining logs if any
+            if (detectCompleted(jobIdentifier)) {
+                sendLogs(webSocket, consoleService.getStreamer(start, jobIdentifier), jobIdentifier);
```

Comment:
Also, the log statement immediately below will likely tell a lie, if you did manage to send the logs across...


Ideally you'd want 3 log statements:

```
Sent ${count} log lines for ${jobId} from ${source} # before the `detectCompleted` block above
Sent ${count} log lines for ${jobId} from ${source} # in the `detectCompleted` block
Sent ${count} log lines for ${jobId} from all sources
```